### ISSUE TITLE
No aarch64 build available

### ISSUE THREAD

**Issue by mlehtima** [M4rtinK/modrana#289]
On devices with Sailfish OS with aarch64 architecture openrepos app installs the old 0.55.20 version which was still noarch but that is too old to work in any recent Sailfish OS version so an aarch64 build of the latest release is needed.

**Comment by mlehtima** [comment]
Just to note that I made aarch64 build myself and it works fine on Xperia 10 II

**Comment by M4rtinK** [comment]
Yeah, I know about this and hopefuly a new build with a aarch64 should be
done in the next few days. Hopefully my 10 II should be up and running with
SFOS by then as well.

And thanks for the report and testing! :)

Dne ne 4. 7. 2021 13:38 uživatel Matti Lehtimäki ***@***.***>
napsal:

**Comment by Wunderfitz** [comment]
I'd also be interested in building the project myself. Where can I find step-by-step build instructions?

**Comment by M4rtinK** [comment]
So I've rebuilt 0.57.3 on the Sailfish OS latest AARCH64 target and the RPM seems to work fine on my Xperia 10 II, so I've added it to [OpenRepos](https://openrepos.net/content/martink/modrana-0).

As for Jolla store I think I'll do a couple more fixes first and wait a bit for Feedback from Open Repos and then will ship the next build both to Jolla Store and Open Repos.

**Comment by M4rtinK** [comment]
> I'd also be interested in building the project myself. Where can I find step-by-step build instructions?
That's a good point, I really need to compile something up to date. :P

At the moment the process is pretty tightly coupled with the Mer OBS - I basically create new tarball and spec file, upload it to OBS, which trigger a new build, that shows up here:

http://repo.merproject.org/obs/home:/MartinK:/sailfish:/modrana/sailfish_latest_aarch64/aarch64/

The OBS project can be seen here:

https://build.merproject.org/project/show/home:MartinK:sailfish:modrana

I also have one other separate project that builds the package differently so that it's Jolla Store compatible.

While this is pretty handy and largely automated for me, it's not really straightforward for user who would want to build modRana themselves, so I'll try to compile some user friendly instructions for how to build modRana with the SDK, that should be easier than involving OBS.

BTW, the amount of native code in modRana that actually needs to be compiled is pretty minuscule, basically just the launcher that sets up the viewport & that makes it possible to run modRana via the handy `harbour-modrana` from command line on Sailfish OS.

ModRana also supports being started without the launcher just via `qmlscene` - you can see that on desktop Linux, just make sure PyOtherSide & `qmlscene` are installed, clone the modRana repo and run modrana.py - the modRana GUI should show up right away without the need to compile anything (this is how I do bulk of modRana development BTW).

On Sailfish OS as far as I can tell something is broken - installing `qt5-qtdeclarative-qmlscene` and `qtchooser` packages should be enough to do the same thing, yet the viewport is then all broken. I'll have to investigate some more.

**Comment by mlehtima** [comment]
I just verified that the aarch64 build in openrepos works correctly on my Xperia 10 II. Thanks.

**Comment by takimata** [comment]
> Where can I find step-by-step build instructions?

Here (is what I had to do to compile locally, including workarounds):

```
git clone https://github.com/M4rtinK/modrana.git
cd modrana
sfdk config --push target SailfishOS-4.1.0.24-aarch64
ln -s packaging/ rpm
ssh root@build
# in build VM:
sb2 -t SailfishOS-4.1.0.24-aarch64 zypper in python3-devel rsync
exit # build VM

make VersionFile
sfdk build --define "with_sailfish 1" --define "with_harbour 1"
```

The error `No provider of '/usr/bin/pathfix.py' found.` can be ignored.
The resulting package can be found in the RPMS subdirectory as usual.
